internal/olm/client/status.go

```diff
@@ -95,6 +95,7 @@ func (s Status) HasInstalledResources() (bool, error) {
 	})
 	for _, r := range s.Resources {
 		if r.Resource != nil {
+			log.Infof("Exists olm resource:%v, project:%v, GVK:%v", r.Resource, r.NamespacedName, r.GVK)
```

Comment:
I would suggest removing the log line. We should not log from library-style functions like this because callers should always control output. Instead, callers can use the `String()` method or iterate the Resources themselves to print details with custom formatting.

```suggestion
```

Reply:
@joelanford Thanks for your suggestion! But, I tried it on the OCP cluster yesterday, I have to remove OLM first(`operator-sdk olm uninstall` doesn't work) manually. I only get the `detected existing OLM resources: OLM must be completely uninstalled before installation` error info. It's difficult for me to find which OLM resource still there. I'm familiar with OLM, for other users, I'm sure it's more difficult for them to find it. So, I add this log so that can help me to find the left OLM resource.

Reply:
For install, we [print the full status](https://github.com/operator-framework/operator-sdk/blob/7db9ec2f5fae19506146bbed5ab70d65757a28e8/internal/olm/installer/manager.go#L84) once the operation completes successfully.

I'm wondering if we could refactor that function slightly to ALWAYS print the status and _then_ handle the error (or lack of error). I think that should accomplish what you're looking for. And I agree that would be helpful!

Reply:
@joelanford That's a great idea! I like that. For now, the logs did noisily, so I update it to list all exist OLM resources. Like this, what do you think?
```console
INFO[0008] Fetching CRDs for version "latest"           
INFO[0079] Fetching resources for version "latest"      
INFO[0110] 	Should remove exist OLM resource： Kind:CustomResourceDefinition, Name:operators.operators.coreos.com 
INFO[0110] 	Should remove exist OLM resource： Kind:CustomResourceDefinition, Name:operatorgroups.operators.coreos.com 
INFO[0110] 	Should remove exist OLM resource： Kind:CustomResourceDefinition, Name:installplans.operators.coreos.com 
INFO[0110] 	Should remove exist OLM resource： Kind:CustomResourceDefinition, Name:clusterserviceversions.operators.coreos.com 
INFO[0110] 	Should remove exist OLM resource： Kind:Deployment, Name:olm-operator 
INFO[0110] 	Should remove exist OLM resource： Kind:CustomResourceDefinition, Name:subscriptions.operators.coreos.com 
INFO[0110] 	Should remove exist OLM resource： Kind:ClusterRoleBinding, Name:olm-operator-binding-olm 
INFO[0110] 	Should remove exist OLM resource： Kind:CatalogSource, Name:operatorhubio-catalog 
INFO[0110] 	Should remove exist OLM resource： Kind:OperatorGroup, Name:olm-operators 
INFO[0110] 	Should remove exist OLM resource： Kind:ClusterRole, Name:aggregate-olm-view 
INFO[0110] 	Should remove exist OLM resource： Kind:Deployment, Name:catalog-operator 
INFO[0110] 	Should remove exist OLM resource： Kind:ClusterRole, Name:aggregate-olm-edit 
INFO[0110] 	Should remove exist OLM resource： Kind:Namespace, Name:olm 
INFO[0110] 	Should remove exist OLM resource： Kind:OperatorGroup, Name:global-operators 
INFO[0110] 	Should remove exist OLM resource： Kind:Namespace, Name:operators 
INFO[0110] 	Should remove exist OLM resource： Kind:ClusterServiceVersion, Name:packageserver 
INFO[0110] 	Should remove exist OLM resource： Kind:ServiceAccount, Name:olm-operator-serviceaccount 
INFO[0110] 	Should remove exist OLM resource： Kind:CustomResourceDefinition, Name:catalogsources.operators.coreos.com 
INFO[0110] 	Should remove exist OLM resource： Kind:ClusterRole, Name:system:controller:operator-lifecycle-manager 
FATA[0110] Failed to install OLM version "latest": detected existing OLM resources: OLM must be completely uninstalled before installation 
```